### PR TITLE
feat: expanded Python docs + Rust docs

### PR DESCRIPTION
**Pull request by danking**

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -3,18 +3,60 @@
...
+
+       For end-users looking to read and write Vortex files.
+
+    .. grid-item-card:: The Benchmarks
+       :link: https://bench.vortex.dev/
+
+       Random access, throughput, and TPC-H.
+
+
+Vortex is a fast, extensible, lightweight-compressed, and random-access columnar file format as well
```
Comment (lwwmanning) [inline docs/index.rst]: Vortex is a fast & extensible columnar file format that is based around state-of-the-art research from the database community. It is built around cascading compression with lightweight encodings (no block compression), allowing for both efficient random access and extremely fast decompression. 

Vortex also includes an accompanying in-memory format for these (recursively) compressed arrays, that is zero-copy compatible with Apache Arrow in uncompressed form. Taken together, the Vortex library is a useful toolkit with *compressed* Arrow data in-memory, on-disk, & over-the-wire.

Vortex aspires to succeed Apache Parquet by pushing the Pareto frontier outwards: 1-2x faster writes, 2-10x faster scans, and 100-200x faster random access reads, while preserving the same approximate compression ratio as Parquet v2 with zstd.

Its features include:

File: docs/index.rst [hunk 2]
```diff
@@ -3,18 +3,60 @@
...
+
+- A zero-copy data layout for disk, memory, and the wire.
+- Kernels for computing on, filtering, slicing, indexing, and projecting compressed arrays.
+- Builtin state-of-the-art codecs including FastLanes (integer bit-packing), ALP (floating point),
+  and FSST (strings).
+- Support for custom user-implemented codecs.
+- Support for, but no requirement for, row groups.
+- A read sub-system supporting filter and projection pushdown.
+
+Spiral's flexible layout empowers writers to choose the right layout for their setting: fast writes,
```
Comment (lwwmanning) [inline docs/index.rst]: s/Spiral/Vortex/g

File: docs/file_format.rst [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+The column layout lays out each column of a struct-typed array as a separate sequence of bytes. Each
+column may or may not recursively use a chunked layout. Column layouts permit readers to push-down
+column projections.
+
+The chunked layout lays out an array as a sequence of row chunks. Each chunk may have a different
+size. A chunked layout permits reader to push-down row filters based on statistics which we describe
+later. Note that, if the laid out array is a struct array, each column uses the same chunk
+size. This is equivalent to Parquet's row groups.
+
+The layout: chunked of struct of chunked of flat, is essentially a Parquet layout with row groups in
```
Comment (lwwmanning) [inline docs/file_format.rst]: maybe call out that these are examples

**Comment by danking** [comment]
commentes addressed

**Review by lwwmanning — APPROVED** [review]

**Pull request merged**